test/distributed/_tensor/test_utils.py

```diff
@@ -119,5 +123,82 @@ def test_compute_local_shape_and_global_offset_2D(self):
                 )
 
 
+class TestFSDP2DLocalShardsAndOffsets(DTensorTestBase):
```

Comment:
I think these tests should be in FSDP2 test folder? Or maybe rename the test cases to something like strided_local_shard tests and we can probably leave it in DTensor